frontend/src/routes/_protected/settings/import.tsx

```diff
@@ -1,13 +1,62 @@
 import { createFileRoute } from '@tanstack/react-router'
 import FileUpload from '@/components/general/fileUpload/FileUpload';
 import GeneralStatusCard from '@/components/general/cards/GeneralStatusCard';
+import Modal from '@/components/general/form/Modal';
+import PrimaryButton from '@/components/general/buttons/PrimaryButton';
+import { useState } from 'react';
 
 export const Route = createFileRoute('/_protected/settings/import')({
   component: ImportFile,
 })
 
 function ImportFile() {
-  const url = '/api-local/v1/import/csv'
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [file, setFile] = useState<File | null>(null);
+  const [message, setMessage] = useState("");
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files && event.target.files.length > 0) {
+      setFile(event.target.files[0]);
+    }
+  };
+
+  const handleConfirm = async () => {
+    setIsModalOpen(false);
+
+    try {
+      if (!file) return;
+
+      setMessage("");
+      const formData = new FormData();
+      formData.append("file", file);
+
+      // TODO: Send form to provided endpoint of backend client
+
+      setFile(null);
+      setMessage("Es wurden erfolgreich neue Daten importiert.");
+    } catch (error) {
+      handleAxiosError(error);
+    }
+  };
+
+  const handleAxiosError = (error: any) => {
+    const { response, message } = error;
+
+    if (response?.data?.error) {
+      const { error: errorMessage, err: additionalMessage } = response.data;
+      const fullErrorMessage = additionalMessage ? `${errorMessage}: ${additionalMessage}` : errorMessage;
+
+      setMessage('Leider ist folgender Fehler aufgetreten: ' + fullErrorMessage);
+    } else if (message) {
+      setMessage('Leider ist folgender Fehler aufgetreten: ' + message);
+    } else {
+      const message = 'Leider ist ein unbekannter Fehler aufgetreten. Bitte wenden Sie sich an eine:n Systemadministrator:in.';
+      setMessage(message);
+    }
+  };
```

Comment:
I would prefer to use the query client here with the functions coming from the backend client. That way we don't have to deal with error messages, load state and so on ourselves.